apps/web/src/lib/isEnsVerified.ts

```diff
@@ -0,0 +1,23 @@
+import type { Profile } from 'lens';
+
+/**
+ *
+ * @param profile - Profile id
+ * @returns `true` if handle is same as ENS name
+ *
+ * Examples for handle "vitalik.lens":
+ *  - ENS: "vitalik.eth" returns `true`
+ *  - ENS: "lens.vitalik.eth" returns `true`
+ *  - ENS "anothervitalik.eth" returns `false` (trying to impersonate)
+ *  - ENS "vitalik.eth.somebodyelse.eth" returns `false` (trying to impersonate)
+ */
+function isEnsVerified(profile: Profile): boolean {
```

Comment:
Hi Pavel, I would like to open the discussion on the condition of this function.

In my opinion, the ENS verified batch should not be dependent on whether ENS and lens handles do match. 

You would be excluded from the functionality if:
- someone squats your ENS handle on lens
- you loose/change your lens handle

It should show up for every user, whose address resolves to an ENS name.

wdyt?

Reply:
Hi @faustofren .. thanks for the feedback

1. To be able to identify if somebody is squating your ENS handle, it would mean that we would need to do additional "search" every time we are creating the ENS/lens identifiers. For example: If you open one user profile, we would need to look for other people based on your LENS handle. Even worse, when displaying a Timeline feed, we would need to do a lot of these searches which would create unnecessary load for APIs. Of course, there would be a possibility to create our own database where we would store something like `isThisUserSquatting: bool` but that would require creating such a database and it would defeat a bit the purpose of having all these data available in native Lens protocol.

2. We should not treat the "ENS name" somehow superior, above everything else. If you own some ENS name, it doesn't necessarily mean that you are entitled to such handle / string in every single naming system / social media. It would rather defeat the purpose of decentralization. But, of course, because of it's history, adoption, popularity; many people think of ENS just like that.

Reply:
Hi Pavel,

1. Agree, don´t think any specific application side is necessary here.

2. Agree, it´s not superior above everything else. We just have to view each level of verification as it´s own and prioritize them.

The ENS name indeed actually verifies your **pseudonymous** identity, including your Ethereum on-chain activity (DAO participation, NFT collecting, Defi participation etc.) up to the past 5 years.

Vitalik could have actually chosen a completely random lens handle. The ENS name verification still would have ensured that this account actually is controlled by Vitalik himself, due to the provenance and significance of his ENS name. No other verification would have been required actually.

The ENS name badge could also be helpful for crypto/web3 sub-communities (DAOs, projects etc.) who are mostly operating pseudonymously to discover and verify each other, without the need to hold a 1:1 ENS/lens handle copy.

wdyt?

BR,
fausto

Reply:
@faustofren I agree with both points and verified ENS badge is in places like `Timeline` and `Profile` should definitely help people from communities identify each other easier. I think we should display the ENS name in `Timeline` (currently it's not). It's already displayed in `Profile` page.

Reply:
also to add, matching ens name and lens handle is logically not viable because of the different length requirements of the two handles.

- ENS: minimum 3 characters
- lens: minimum 5 characters